projects/packages/my-jetpack/_inc/components/product-interstitial/PricingInterstitial.jsx

```diff
@@ -0,0 +1,347 @@
+	);
+
+	const handleGetProduct = useCallback( () => {
+		setLoadingButton( 'paid' );
+		trackProductOrBundleClick( { ctaText: config?.tiers?.paid?.cta } );
+		clickHandler( { checkout: paidCheckoutRun, product: detail, tier: 'paid' } );
+	}, [
+		trackProductOrBundleClick,
+		clickHandler,
+		paidCheckoutRun,
+		detail,
+		config?.tiers?.paid?.cta,
+	] );
+
+	const handleGetBundle = useCallback( () => {
+		if ( config?.bundle ) {
+			setLoadingButton( 'bundle' );
+			trackProductOrBundleClick( {
+				customSlug: config.bundle,
+				ctaText: config?.tiers?.bundle?.cta,
+			} );
+			clickHandler( { checkout: bundleCheckoutRun, product: bundleDetail, tier: 'bundle' } );
+		}
+	}, [ trackProductOrBundleClick, clickHandler, bundleCheckoutRun, bundleDetail, config ] );
+
+	const handleFreeActivation = useCallback( () => {
+		setLoadingButton( 'free' );
+		trackProductOrBundleClick( { isFreePlan: true, ctaText: config?.tiers?.free?.cta } );
+		clickHandler( { checkout: null, product: detail, tier: 'free' } );
+	}, [ trackProductOrBundleClick, clickHandler, detail, config?.tiers?.free?.cta ] );
+
+	// If no config exists, fallback to old ProductInterstitial
+	if ( ! config ) {
+		return <ProductInterstitial slug={ slug } installsPlugin={ true } />;
+	}
+
+	// Handle tiered pricing like trunk does - check for tiers.upgraded first
+	const productPricing = detail?.pricingForUi?.tiers?.upgraded
+		? {
+				...detail.pricingForUi.tiers.upgraded,
+				// Calculate monthly prices from annual if needed
+				fullPricePerMonth: detail.pricingForUi.tiers.upgraded.fullPrice / 12,
+				discountPricePerMonth: detail.pricingForUi.tiers.upgraded.discountPrice / 12,
+		  }
+		: detail?.pricingForUi;
+	const bundlePricing = bundleDetail?.pricingForUi;
+
+	// Get currency code with USD fallback
+	const currencyCode = productPricing?.currencyCode || bundlePricing?.currencyCode || 'USD';
+
+	return (
+		<AdminPage
+			showHeader={ false }
+			showBackground={ false }
+			useInternalLinks={ shouldUseInternalLinks() }
+		>
+			<Container
+				className={ styles.interstitialContainer }
+				horizontalSpacing={ 3 }
+				horizontalGap={ 3 }
+			>
+				<Col className={ styles[ 'product-interstitial__header' ] }>
+					<GoBackLink onClick={ onClickGoBack } />
+				</Col>
+				<Col>
+					<PricingTable
+						title={ config.title }
+						items={ config.features }
+						showIntroOfferDisclaimer={ false }
+						headerLogo={ config.logo ? <config.logo height={ 32 } /> : null }
+					>
+						<PricingTableColumn>
+							<PricingTableHeader title={ config.tiers.free.name }>
+								<ProductPrice
+									price={ 0 }
+									legend=""
+									currency={ currencyCode }
+									hidePriceFraction
+									variant="simple"
+								/>
+								<Button
+									fullWidth
+									variant="secondary"
+									onClick={ handleFreeActivation }
+									isLoading={ loadingButton === 'free' }
+									disabled={ loadingButton && loadingButton !== 'free' }
```

Comment:
I think we should disable all the CTAs when one is clicked. Thus, all of these can be like this:

```tsx
disabled={ Boolean( loadingButton )  }
```

Otherwise, I can click on two buttons subsequently.

Reply:
Thanks, done in https://github.com/Automattic/jetpack/commit/6d60d2c0d2d668f882a08ba671f112702c70a81f!